packages/dds/tree/src/core/tree/detachedFieldIndex.ts

```diff
@@ -179,6 +309,63 @@ export class DetachedFieldIndex {
 		this.rootIdAllocator = idAllocatorFromMaxId(
 			detachedFieldIndex.maxId,
 		) as IdAllocator<ForestRootId>;
-		this.detachedNodeToField = detachedFieldIndex.data;
+
+		this.detachedNodeToField = new Map();
+		this.latestRelevantRevisionToFields = new Map();
+		this.fullyLoaded = false;
+		const rootMap = new Map<ForestRootId, Delta.DetachedNodeId>();
+		forEachInNestedMap(detachedFieldIndex.data, (root, major, minor) => {
+			setInNestedMap(this.detachedNodeToField, major, minor, {
+				root,
+				latestRelevantRevision: fakeRevisionWhenNotSet,
+			});
+			rootMap.set(root, { major, minor });
+		});
+
+		this.latestRelevantRevisionToFields.set(fakeRevisionWhenNotSet, rootMap);
+	}
+
+	/**
+	 * Sets the latest relevant revision for any roots that have an undefined latest relevant revision.
+	 * This occurs when the detached field index is loaded from a summary and can only be called once after
+	 * the summary has been loaded.
+	 */
+	public setRevisionsForLoadedData(latestRevision: RevisionTag): void {
+		assert(
+			!this.fullyLoaded,
+			"revisions should only be set once using this function after loading data from a summary",
+		);
+
+		const rootMap = new Map();
+		forEachInNestedMap(this.detachedNodeToField, (entry, major, minor) => {
+			if (entry.latestRelevantRevision === fakeRevisionWhenNotSet) {
```

Comment:
Is `entry.latestRelevantRevision === fakeRevisionWhenNotSet` ever false?

Reply:
nope :)